src/rpc.rs

```
@@ -19,11 +19,14 @@ where
   T: Send + std::fmt::Debug + 'static,
 {
   let (tx, rx) = channel::<T>();
-  thread::spawn(move || {
+
+  let _jh = thread::spawn(move || {
```

maybe `thread::spawn(move || -> Result<()> {` and then `Ok(tx.send(r)?)` ?

---

this could be an option, what would be the advantage?

---

We could bubble up the error up to the caller and let them decide what to do with the error, either panic, retry, or something else.

---

in this case we are ignoring the JoinHandle and the result of the sync_code is pushed over the oneshot. Adding a result would not change anything in the flow. Lets think about how we can get this more robust, maybe requiring the sync_code function signature to return a Result and then do something with the JoinHandle

---

Sure, I think its such a low-probability scenario that the one-shot will fail that its not critical to handle this error in a more elaborate way.